### ISSUE TITLE
Remove unfinished uploads

### ISSUE THREAD

**Issue by ck23onGithub** [VeliovGroup/Meteor-Files#267]
Question:
Is it possible to get the name of the file in the FS when the upload starts?

it immediately creates the file in FS when upload is started, right? is there any possible way to get this?

I'm thinking of deleting the file if the upload was not finished.

Why I need this? Here is the scenario I'm thinking:
1. Client uploads a file
2. While upload is ongoing, he closed his browser (or something like losing connection from the client side) and returns only after a few hrs
3. A file remains in the FS and I need to delete it

I'm thinking if there is a way to get the file name of the created file, from the client side, I'll just pass it to the server and then remove it some time in the future.

Thank you

**Comment by dr-dimitru** [comment]
It's is already done for you.
read docs for `continueUploadTTL` passed to [`Constructor`](https://github.com/VeliovGroup/Meteor-Files/wiki/Constructor)

**Comment by ck23onGithub** [comment]
ok. let me check this again. i might be missing something in the tests that I'm doing.

**Comment by ck23onGithub** [comment]
i tried
`continueUploadTTL: 180`
i closed the tab while the upload is going on. then i closed the meteor app after 5min (2min after the `continueuploadTTL` expires). the file is still in the FS.

is there something wrong with my app or I'm expecting the wrong thing?

**Comment by dr-dimitru** [comment]
>  i closed the meteor app

Do you mean shut down the server?

**Comment by ck23onGithub** [comment]
I'm in development stage as of now, so i think thats the same as shutting down the server.

with other processes before, if the file is expected to be deleted, it is being deleted when i close the meteor app. that's why i'm expecting for this to do the same.

**Comment by dr-dimitru** [comment]
Shutdown the server is same if you stop the `meteor` process.
You're right. I cached the code, yes unfinished uploads is not removed.
I'll add this feature to the next release

**Comment by ck23onGithub** [comment]
thanks! and i hope others may find this useful as well.

**Comment by dr-dimitru** [comment]
@ck23onGithub honestly I thought it was implemented.
Thank you for your persistence.

**Comment by dr-dimitru** [comment]
Published in [v1.7.6](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.7.6)
